### PR TITLE
4691 support block type strings in resnet and senet

### LINKED ISSUE #4691: support block type strings in resnet
**Is your feature request related to a problem? Please describe.**
from https://github.com/Project-MONAI/MONAI/discussions/4678, the resnet input args should be able to understand strings in addition to the class types, the corresponding docstrings should be revised https://github.com/Project-MONAI/MONAI/blob/d99857cb49aab464ac38f7926ce4c6f299417721/monai/networks/nets/resnet.py#L173-L175

### PR DESCRIPTION
**Pull request by KumoLiu**
Signed-off-by: KumoLiu <yunl@nvidia.com>

Fixes #4691 #4678 #4701 .

### Description
support block type strings in resnet and senet.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/networks/nets/resnet.py [hunk 1]
```diff
@@ -150,6 +150,10 @@ class ResNet(nn.Module):
 
     Args:
         block: which ResNet block to use, either Basic or Bottleneck.
+            ResNet block class or str.
+            for Basic: ResNetBlock or 'basic'
+            for Bottleneck: ResNetBottleneck or 'bottleneck'
+        Options are 'basic' and 'bottleneck'.
```
Comment (yiheng-wang-nv) [inline monai/networks/nets/resnet.py:156]: this line seems can be removed.

**Review by yiheng-wang-nv — APPROVED** [review]
LGTM

**Review by Nic-Ma — APPROVED** [review]
Look good to me.
@wyli Do you have any other comments?

Thanks.

**Pull request merged**